conans/client/loader.py

```diff
@@ -70,8 +71,11 @@ def __init__(self, runner, settings, profile):
         @param cached_env_values: EnvValues object
         """
         self._runner = runner
+
+        assert isinstance(settings, Settings)
         settings.values = profile.settings_values
-        assert settings is None or isinstance(settings, Settings)
+        preprocess(settings, output)
```

Comment:
You can use the ``settings`` object, not just the values, to process it, augment it, while keeping its validation rules. Though it is true that it might break if someone is applying settings constraints. But you can apply the logic to the ``profile.settings``